### ISSUE TITLE
[1.5] No keyring error with --export

### ISSUE THREAD

**Issue by heitorPB** [canonical/charmcraft#730]
Hello,

I am trying to export a limited auth token to use with Github actions, but I had no success so far:

```bash
$ CHARMCRAFT_AUTH="" charmcraft login --export=gh.auth --charm slurmd --charm slurmdbd --charm slurmctld --charm slurmrestd --permission package-manage
No keyring found to store or retrieve credentials from.
Full execution log: '/home/h/snap/charmcraft/common/cache/charmcraft/log/charmcraft-20220325-145605.991757.log'

$ cat /home/h/snap/charmcraft/common/cache/charmcraft/log/charmcraft-20220325-145605.991757.log
2022-03-25 14:56:05.992 Starting charmcraft version 1.5.0
2022-03-25 14:56:05.992 Raw pre-parsed sysargs: args={'help': False, 'verbose': False, 'quiet': False, 'trace': False, 'project_dir': None} filtered=['login', '--export=gh.auth', '--charm', 'slurmd', '--charm', 'slurmdbd', '--charm', 'slurmctld', '--charm', 'slurmrestd', '--permission', 'package-manage']
2022-03-25 14:56:05.992 General parsed sysargs: command='login' args=['--export=gh.auth', '--charm', 'slurmd', '--charm', 'slurmdbd', '--charm', 'slurmctld', '--charm', 'slurmrestd', '--permission', 'package-manage']
2022-03-25 14:56:05.992 Couldn't find config file '/home/h/projects/slurm-charms/.github/workflows/charmcraft.yaml'
2022-03-25 14:56:05.993 Command parsed sysargs: Namespace(bundle=None, channel=None, charm=['slurmd', 'slurmdbd', 'slurmctld', 'slurmrestd'], export=PosixPath('gh.auth'), permission=['package-manage'], ttl=None)
2022-03-25 14:56:05.993 System details: OSPlatform(system='ubuntu', release='20.04', machine='x86_64'); Environment: None
2022-03-25 14:56:06.069 No keyring found to store or retrieve credentials from.
2022-03-25 14:56:06.071 Traceback (most recent call last):
2022-03-25 14:56:06.071   File "/snap/charmcraft/834/lib/charmcraft/commands/store/store.py", line 179, in __init__
2022-03-25 14:56:06.071     self._client = Client(charmhub_config.api_url, charmhub_config.storage_url)
2022-03-25 14:56:06.071   File "/snap/charmcraft/834/lib/charmcraft/commands/store/client.py", line 57, in __init__
2022-03-25 14:56:06.071     super().__init__(
2022-03-25 14:56:06.071   File "/snap/charmcraft/834/lib/craft_store/store_client.py", line 84, in __init__
2022-03-25 14:56:06.071     super().__init__(
2022-03-25 14:56:06.071   File "/snap/charmcraft/834/lib/craft_store/base_client.py", line 65, in __init__
2022-03-25 14:56:06.071     self._auth = Auth(
2022-03-25 14:56:06.071   File "/snap/charmcraft/834/lib/craft_store/auth.py", line 103, in __init__
2022-03-25 14:56:06.071     raise errors.NoKeyringError()
2022-03-25 14:56:06.071 craft_store.errors.NoKeyringError: No keyring found to store or retrieve credentials from.
2022-03-25 14:56:06.071 Full execution log: '/home/h/snap/charmcraft/common/cache/charmcraft/log/charmcraft-20220325-145605.991757.log'
```

I am using Ubuntu LTS server 20.04, headless. Charmcraft from latest/candidate (version 1.5.0, rev 834).

I thougth the `--export` flag would cause Charmcraft to not use a keyring?

**Comment by sergiusens** [comment]
This is indeed not the case today, I will discuss with @facundobatista

**Comment by facundobatista** [comment]
@sergiusens it looks like Craft Store is always expecting a keyring to be available when not having a `CHARMCRAFT_AUTH` set (and of course will not allow to use the  `login` command when having that environment variable).

**Comment by sergiusens** [comment]
@facundobatista correct, this is what you asked for, you wanted login to always login and potentially show the credentials.
I brought it up, we need a new key to say that we don't want to record this, either by not passing in the environment variable on init or the ephemeral argument I proposed.

**Comment by facundobatista** [comment]
Fixed by #760.
